src/indicator.js

```diff
@@ -60,36 +60,20 @@ var CPUFreqIndicator = new Lang.Class({
         this.cpufreq = 800;
         this.cpucount = 0;
         this.isTurboBoostActive = true;
-        this.minVal = 0;
-        this.maxVal = 30;
+        this.minVal = this._getMinCheck();
+        this.maxVal = 100;
 
-        global.log('Inside cpupower.CPUFreqIndicator')
-        // read the last-settings file.
-        if(!GLib.file_test(EXTENSIONDIR + '/.last-settings', GLib.FileTest.EXISTS))
+        global.log('Inside cpupower.CPUFreqIndicator');
+        // read the cached settings file.
+        if(GLib.file_test(LASTSETTINGS, GLib.FileTest.EXISTS))
         {
-            let result = GLib.spawn_command_line_sync(CPUFREQCTL + ' turbo get');
-            let returnCode = result[1];
-            this.isTurboBoostActive = returnCode;
-
-            result = GLib.spawn_command_line_sync(CPUFREQCTL + ' min get');
-            returnCode = result[1];
-            this.minVal = returnCode;
-
-            result = GLib.spawn_command_line_sync(CPUFREQCTL + ' max get');
-            returnCode = result[1];
-            this.maxVal = returnCode;
-        }
-        else
-        {
-            let lines = Shell.get_file_contents_utf8_sync(EXTENSIONDIR + '/.last-settings').split('\n');
+            let lines = Shell.get_file_contents_utf8_sync(LASTSETTINGS).split('\n');
             if(lines.length > 2)
             {
                 this.minVal = parseInt(lines[0]);
                 this.maxVal = parseInt(lines[1]);
                 this.isTurboBoostActive = (lines[2].indexOf('true') > -1);
-                this._updateMax(true);
```

Comment:
Removing these lines breaks functionality imho. The code was intended to restore the settings used during the last execution.

The idea was basically to use the settings from the file if it exists, otherwise initialize with the current settings.